libr/bin/format/elf/elf.c

```diff
@@ -2504,27 +2507,32 @@ int Elf_(r_bin_elf_is_big_endian)(ELFOBJ *bin) {
 
 /* XXX Init dt_strtab? */
 char *Elf_(r_bin_elf_get_rpath)(ELFOBJ *bin) {
-	char *ret = NULL;
-	int j;
+	char *ret;
+	Elf_(Xword) val;
 
-	if (!bin || !bin->phdr || !bin->dyn_buf || !bin->strtab) {
+	if (!bin || !bin->phdr || !bin->strtab) {
 		return NULL;
 	}
-	for (j = 0; j< bin->dyn_entries; j++) {
-		if (bin->dyn_buf[j].d_tag == DT_RPATH || bin->dyn_buf[j].d_tag == DT_RUNPATH) {
-			if (!(ret = calloc (1, ELF_STRING_LENGTH))) {
-				perror ("malloc (rpath)");
-				return NULL;
-			}
-			if (bin->dyn_buf[j].d_un.d_val > bin->strtab_size) {
-				free (ret);
-				return NULL;
-			}
-			strncpy (ret, bin->strtab + bin->dyn_buf[j].d_un.d_val, ELF_STRING_LENGTH);
-			ret[ELF_STRING_LENGTH - 1] = '\0';
-			break;
-		}
+
+	if (bin->dyn_info.dt_rpath != ELF_XWORD_MAX)  {
+		val = bin->dyn_info.dt_rpath;
+	} else if (bin->dyn_info.dt_runpath != ELF_XWORD_MAX){
+		val = bin->dyn_info.dt_runpath;
+	} else {
+		return NULL;
 	}
+
+	if (val > bin->strtab_size) {
+		return NULL;
+	}
+
+	if (!(ret = calloc (1, ELF_STRING_LENGTH))) {
+		perror ("malloc (rpath)");
```

Comment:
Remove thiis perror, theres no need to show an error when the heap cant feed us with love chunks